### PR TITLE
Add --use-static-frameworks lint option

### PR DESCRIPTION
**Pull request by paulb777**
Also fix a bad merge in the CHANGELOG.md

### CODE REVIEW

**Comment by paulb777** [comment]
Rebased

**Comment by dnkoutso** [comment]
I continue to feel that the proliferation of these options is making things much worse for maintainers imo but I don't see a clear way to do manage this effectively.

Basically an author can now publish a pod under very specific set of options (modular headers, use libraries and now static frameworks) which means a consumer of a pod must somehow know that the pod works only works in a specific way.

Do you have any thoughts or ideas about this?

**Comment by paulb777** [comment]
@dnkoutso I'd be fine to remove the option from push.

The reason I want this option is that the boat has already sailed on these options available in Podfile's and it would be a nice for authors to have an easy way to set up CI to test all the options before publishing.  

Specifically, I'd like to use it here for example - https://github.com/firebase/firebase-ios-sdk/blob/master/.github/workflows/storage.yml#L95

File: CHANGELOG.md [hunk 1]
```diff
@@ -91,10 +99,6 @@ To install release candidates run `[sudo] gem install cocoapods --pre`
 
 ##### Enhancements
 
-* Option to lint a specified set of test_specs
```
Comment (amorde) [inline CHANGELOG.md:94]: is this correct?

Comment (paulb777) [reply CHANGELOG.md:94]: Yes, #9392 is in master and not 1.9.1

**Pull request merged**